src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,19 +60,26 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return new String(block.getBlock()+ADDRESS_SPLIT_POINT
+        		+street.getStreet()+ADDRESS_SPLIT_POINT
+        		+unit.getUnit()+ADDRESS_SPLIT_POINT
+        		+postalCode.getPostalCode());
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && getAddress().equals(((Address) other).getAddress())); // state check
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        return new String(block.getBlock()+ADDRESS_SPLIT_POINT
+        		+street.getStreet()+ADDRESS_SPLIT_POINT
```

Comment:
The implementation of the hashCode() does not look right. hashCode() of a class shouldn't be the hashcode of a String object. This is not correct.